### DISCUSSION TITLE
In what folder/files are the vehicles info stored in DDT4ALL

### DISCUSSION THREAD

**Discussion by AndyIreland** [cedricp/ddt4all#962] [Q&A]
I know where the ecu database is (ecu.zip) but what folder or file are the vehicles stored in - its not in the 'vehicles' folder because I have looked in there and its empty. 
![image](https://github.com/cedricp/ddt4all/assets/49172217/b6bbba4b-c5b4-4b11-9202-a8eb282e3565)

![image](https://github.com/cedricp/ddt4all/assets/49172217/ec23b443-5031-4c78-a17e-adf3ce2a8a43)

**Comment by KarelSvo** [comment]
https://github.com/cedricp/ddt4all/blob/winXP/ddt4all.py#L38

Here is the vehicle list.
Back then it was still sorted sensibly.
Not like in the current Py3 version, where the list starts with many vehicle types that 99.9% of users never use.
This list is a 1:1 copy of the ddt2000 vehicle list and sorted alphabetically.

I created a stable Py2 version and edited the type list sensibly.
The most used vehicle types  the beginning the list and type designations changed to real sales designations.

**Reply by AndyIreland** [reply]
thanks, tried it - but browsing through quickly there is no Zoe Ph2 and maybe a few more vehicles missing possibly from the dtt4all.py from the 'all cars' drive2 version with this ddt4all.py file (I have renamed the other one ddt4all.pyold)

**Reply by KarelSvo** [reply]
@AndyIreland  I sent a message to your pc.repair.coolaney address.

**Comment by Furtif** [comment]
the empty vehicles folder is normal and is recreated if you delete it to avoid path errors, this folder is used with the DTT2000 Real DataBase and not with zip, in any case you can use this folder for certain dependencies such as generating dumps with [vehicles/GenericAddressing.xml](https://github.com/cedricp/ddt4all/files/13693709/GenericAddressing.zip)

## Command lines (uses python3 in mac/linux):
 - ``` python parameters.py --dumpaddressing``` (generate address/addressing.json)
 - in pr opened https://github.com/cedricp/ddt4all/pull/963  ``` python parameters.py --dumpsnat``` (generate address/snat.json) and  ``` python parameters.py --dumpdnat``` (generate address/dnat.json)

**Comment by Furtif** [comment]
@KarelSvo you have to stop with python 2.7 you installed it at the time you have the dependencies ok python 2.x no longer has any support format the pc and try to install the dependencies to see if you can.

**Comment by KarelSvo** [comment]
> not in the 'vehicles' folder

https://github.com/cedricp/ddt4all/blob/winXP/ddt4all.py#L368
https://github.com/cedricp/ddt4all/blob/winXP/ddt4all.py#L605


When you scan a vehicle or manually select Ecus,
you can save the list of these ECUs in an .ecu file using the File/Save ECU list function and reload it later.
This .ecu file is stored in the ddt4all/vehicles folder.

**Comment by KarelSvo** [comment]
> you have to stop with python 2.7

I only use Py2 versions of ddt4all and Pyren as they are backwards compatible with my XP diagnostic laptop and function stably.

I find it a bug that, that was changed from Py2 to Py3 in the master branch.
For Pyren, the versions have been separated and can be updated separately. This supports the needs and technical requirements of all users.

**Reply by Furtif** [reply]
ok but nowadays everyone uses win10 or higher linux latest versions and macos the same, a pc with windows 10 in 2024 will be a very old pc as support with microsoft ends, if not you have already tested the pr or branch https://github.com/cedricp/ddt4all/pull/963 or https://github.com/cedricp/ddt4all/tree/add_dump

**Comment by KarelSvo** [comment]
> but nowadays everyone uses ...

Start a survey in a large vehicle forum about which operating system diagnostic software is used on and you will see that the reality is different.
